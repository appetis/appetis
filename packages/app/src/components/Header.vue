<template>
  <div class="header">
    <div class="header-title">David Health</div>
    <div class="header-menus">
      <vue-feather size="18" type="search" class="header-icons" id="header-icon-search"></vue-feather>
      <vue-feather
        size="18"
        type="message-square"
        class="header-icons"
        id="header-icon-message"
      ></vue-feather>
      <vue-feather size="18" type="bell" class="header-icons" id="header-icon-bell"></vue-feather>
      <img :src="avatar" class="header-avatar" />
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      avatar: require('@/assets/images/avatar.png'),
    }
  },
}
</script>
