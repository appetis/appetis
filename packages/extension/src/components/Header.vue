<template>
  <div class="header">
    <div class="header-title">HW Project (Workly)</div>
    <div class="header-menus">
      <v-icon
        name="search"
        base-class="v-icon"
        class="header-icons"
        id="header-icon-search"
      ></v-icon>
      <v-icon
        name="message-square"
        base-class="v-icon"
        class="header-icons"
        id="header-icon-message"
      ></v-icon>
      <v-icon
        name="bell"
        base-class="v-icon"
        class="header-icons"
        id="header-icon-bell"
      ></v-icon>
      <div class="ml-4 profile-photo profile-online">
        <img
          :src="avatar"
          class="header-avatar"
          @click="openProfile()"
          id="header-profile"
        />
      </div>
    </div>
    <Profile
      v-show="showProfile"
      @close="closeProfile"
      :avatar="avatar"
      id="modal-profile"
    />
  </div>
</template>

<script>
import Profile from './user/Profile'
export default {
  components: {
    Profile,
  },
  data() {
    return {
      avatar: require('@/assets/images/avatar.png'),
      showProfile: false,
    }
  },
  methods: {
    openProfile() {
      this.showProfile = true
    },
    closeProfile() {
      this.showProfile = false
    },
  },
}
</script>
