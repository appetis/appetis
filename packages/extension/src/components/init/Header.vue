<template>
  <div class="modal-header">
    <img :src="require('@/assets/images/w.png')" alt="worlky small logo">
    <div class="modal-title">{{ title }}</div>
  </div>
</template>
<script>
export default {
  name: 'init_header',
  props: {
    title: {
      type: String,
      required: true
    }
  }
}
</script>