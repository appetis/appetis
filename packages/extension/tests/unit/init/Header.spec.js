import Header from '@/components/init/Header'
import { mount } from '@vue/test-utils'

const data = {
  title: 'Login',
}
const factory = () => {
  return mount(Header, {
    propsData: data,
  })
}
describe('init/Header', () => {
  it('code modal snapshot', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.get('.modal-title').text()).toBe(data.title)
  })
})
