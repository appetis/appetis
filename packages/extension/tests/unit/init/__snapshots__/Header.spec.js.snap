// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init/Header code modal snapshot 1`] = `
<div class="modal-header"><img src="" alt="worlky small logo">
  <div class="modal-title">Login</div>
</div>
`;
